app/javascript/app/ssn-field.js

```diff
@@ -0,0 +1,47 @@
+import Cleave from 'cleave.js';
+
+const { I18n } = window.LoginGov;
+
+/* eslint-disable no-new */
+function formatSSNField() {
+  const inputs = document.querySelectorAll('input.ssn-toggle[type="password"]');
+
+  if (inputs) {
+    [].slice.call(inputs).forEach((input, i) => {
+      const el = `
+        <div class="mt1 right">
+          <label class="btn-border" for="ssn-toggle-${i}">
+            <div class="checkbox">
+              <input id="ssn-toggle-${i}" type="checkbox">
+              <span class="indicator"></span>
+              ${I18n.t('forms.ssn.show')}
+            </div>
+          </label>
+        </div>`;
+      input.insertAdjacentHTML('afterend', el);
+
+      let ssnCleave;
+      const toggle = document.getElementById(`ssn-toggle-${i}`);
+      toggle.addEventListener('change', function () {
+        input.type = toggle.checked ? 'text' : 'password';
+
+        if (!toggle.checked) {
+          if (ssnCleave) {
+            ssnCleave.destroy();
+          }
+
+          input.value = input.value.replace(/-/g, '');
+        } else {
+          ssnCleave.destroy();
```

Comment:
To the previous comment about optional chaining, it also appears that on initial load, since the field is hidden, there is no `ssnCleave` instance, so this errors when toggling the checkbox.

```
Uncaught TypeError: Cannot read property 'destroy' of undefined
```

The optional chaining would resolve this. But there's another point as well, which is that we're making assumptions about the default state, and assuming that cleave only needs to be initialized on change. But strictly speaking, it needs to be initialized any time the field is visible. So in a future scenario where we'd decide to show the text by default, we'd not be initializing it.

There's some prior art at #4157 to addressing this.

Reply:
I hacked around a bit and am in new territory, and I'm not sure if I'm handling the potential for multiple inputs correctly, but I added daf511f to address the initialization and syncing.

Reply:
Looks good. We could also inline the function definition of `sync` so it has direct access to the variables in the outer scope, in order to avoid passing as arguments.

```js
let ssnCleave;

function sync() {
  input.type = toggle.checked ? 'text' : 'password';
  input.value = input.value.replace(/-/g, '');

  ssnCleave?.destroy();
  ssnCleave = toggle.checked
    ? new Cleave(input, {
        numericOnly: true,
        blocks: [3, 2, 4],
        delimiter: '-',
      })
    : null;
}

toggle.addEventListener('change', sync);
sync();